modules/machineset-gcp-confidential-vm.adoc

```diff
@@ -0,0 +1,55 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating_machinesets/creating-machineset-gcp.adoc
+// * machine_management/control_plane_machine_management/cpmso-using.adoc
+
+ifeval::["{context}" == "cpmso-using"]
+:cpmso:
+endif::[]
+
+:_content-type: PROCEDURE
+[id="machineset-gcp-confidential-vm_{context}"]
+= Configuring Confidential VM by using machine sets
+
+By editing the machine set YAML file, you can configure the Confidential VM options that a machine set uses for machines that it deploys.
+
+For more information about Confidential Compute features, functionality, and compatibility, see the GCP Compute Engine documentation about link:https://cloud.google.com/compute/confidential-vm/docs/about-cvm[Confidential VM].
+
+.Procedure
+
+. In a text editor, open the YAML file for an existing machine set or create a new one.
+
+. Edit the following section under the `providerSpec` field:
++
+[source,yaml]
+----
+ifndef::cpmso[]
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+endif::cpmso[]
+ifdef::cpmso[]
+apiVersion: machine.openshift.io/v1
+kind: ControlPlaneMachineSet
+endif::cpmso[]
+...
+spec:
+  template:
+    spec:
+      providerSpec:
+        value:
+          confidentialCompute: Enabled <1>
+          onHostMaintenance: Terminate <2>
+          machineType: n2d-standard-8 <3>
+...
+----
+<1> Specify whether Confidential VM is enabled. Valid values are `Disabled` or `Enabled`.
+<2> #not sure what this param is doing#
```

Comment:
Hey @eranco74 :) What does `onHostMaintenance` do?

Reply:
Determine the behavior of the VM when a host maintenance event (GCE needs to stop the VM to perform a hardware or software update) occurs.
If `onHostMaintenance` is set to `Migrate` GCE will live migrate the VM to a new host, instead of stopping it.
In case of confidential VM live migration in case of maintenance event isn't supported.
  
https://cloud.google.com/compute/docs/instances/setting-vm-host-options
https://github.com/openshift/api/blob/d964c7a58d7554af90a86f64d7f83a4cdc3f6351/machine/v1beta1/types_gcpprovider.go#L129